File: G6 Автостоянка на улице Кантора/matv864.py
# простите, что сделал это, но на плюсах не залетало
# решение объяснил в C
# удалять это - поздно

def print_coords(x):
    n_left = ((8 * x + 1) ** 0.5 - 3) / 2
    # n_right = (math.sqrt(8 * x + 1)-1)/2
    n = int(n_left) + 1
    x_left = (n * (n + 1)) // 2
    # x_right = ((n + 1) * (n + 2)) // 2
    print(x_left)
    t = x - x_left
    return n - t, t
    
    
with open("input.txt", "r") as inp, open("output.txt", "w") as out:
    input_text = inp.read().split()
    n = int(input_text[0])
    
    for i in input_text[1:]:
        x = int(i)
        x, y = print_coords(x)
        out.write(f"{x} {y}\n")

File: G6 Автостоянка на улице Кантора/matv864.c
/*
объясняю это нечто
я представил всё в виде диагоналей, в которых элементы возрастают на +1
и по сути мне нужно сначала найти нужную диагональ
я смотрю на первую строку и пытаюсь найти место, где left <= x <= right
(((
    нахожу я это вот как
    каждый элемент первой строки это f(x) = ( n * (n + 1) ) / 2
    и через систему из 2 неравенств нахожу формулу
)))
а затем зная первый элемент диагонали и его координату высчитываю итоговые координаты

*/


#include <stdio.h>
#include <math.h>

typedef struct Coords Coords;
struct Coords{
    long long int x, y;
};


Coords get_coords(long long int x){
    Coords result;
    long long int n_left, n, x_left, t;
    n_left = (int)((sqrt(8 * x + 1) - 3)/2);
    n = floor(n_left);
    n++;
    x_left = (n * (n + 1)) / 2;
    t = x - x_left;
    
    result.x = n - t;
    result.y = t;
    return result;
}


int main(){
    FILE *inp = fopen("input.txt", "r");
    FILE *out = fopen("output.txt", "w");
    
    Coords result;
    
    
    long long int n, x;
    fscanf(inp, "%lld", &n);
    while(n--){
        fscanf(inp, "%lld", &x);
        result = get_coords(x);
        fprintf(out, "%lld %lld\n", result.x, result.y);
    }

    return 0;
}